### PR TITLE
Check if HTTP status for new url is 200 on Obama archive.

### PR DESCRIPTION
**Pull request by liyanchang**

### CODE REVIEW

**Comment by liyanchang** [comment]
FYI @robbiethegeek had to recreate the PR as I transferred my repo to Digital44.

**Comment by robbiethegeek** [comment]
Happy to submit it to your repo

File: src/main.js [hunk 1]
```diff
@@ -6,12 +6,16 @@ chrome.webRequest.onCompleted.addListener(function(details) {
...
                     var newUrl = new URL(details.url);
                     newUrl.host = "obamawhitehouse.archives.gov";
-
-                    // Redirect
-                    chrome.tabs.update(tabs[0].id, {url: newUrl.href});
+                    var xhr = new XMLHttpRequest();
+                    // Had to use a CORS proxy - https://crossorigin.me/
+                    xhr.open("GET", `https://crossorigin.me/${newUrl.href}`, false);
+                    xhr.send()
+                    if (xhr.status == 200) {
```
Comment (liyanchang) [inline src/main.js]: What happens if the crossorigin.me returns a 500? or times out?

**Review by liyanchang — COMMENTED** [review]
Just would like us to consider the failure cases of the new dependency.

**Comment by robbiethegeek** [comment]
Done @liyanchang.

**Comment by liyanchang** [comment]
So my tests show me that it takes 25+ seconds to either get a 200 or 30 seconds for a 522 cloudflare error. That means 25+ seconds of waiting before the obama archive loads if it does exist. Can I challenge us to think about how to do this without a 25+ second user visible load time?

**Comment by liyanchang** [comment]
Closing as I haven't been able to get this to work consistently. Free free to reopen if others are.

**Pull request closed without merging**